ingestion/src/metadata/sampler/sqlalchemy/sampler.py

```diff
@@ -199,25 +199,12 @@ def fetch_sample_data(self, columns: Optional[List[Column]] = None) -> TableData
                 if col.name != RANDOM_LABEL and col.name in names
             ]
         with self.get_client() as client:
-            try:
```

Comment:
The `try` / `catch` offuscate some error -- unsure why we introduced this.

e.g. passing a `MONTH` partition on a BQ timestamp will throw an error (as only DAY and HOUR can be used), which will be caught by the `Exception` which in turn will run the query against the raw dataset and the user will see a `partition field required ...` error message (instead of the correct one)